diff --git a/app/constants.mjs b/app/constants.mjs
index d6a9c3c..155b775 100644
--- a/app/constants.mjs
+++ b/app/constants.mjs
@@ -9,5 +9,14 @@ export const assetBaseUrl = "/assets/";
  * @type {import("@webstudio-is/image").ImageLoader}
  */
 export const imageLoader = ({ src }) => {
-  return src;
+  // image paths already have the "/assets/" prefix, so we don't include the assetBaseUrl here.
+  return import.meta.env.BASE_URL === "/"
+      ? src
+      : import.meta.env.BASE_URL + src;
 };
+
+export const assetLoader = (src) => {
+  return import.meta.env.BASE_URL === "/"
+      ? assetBaseUrl + src
+      : import.meta.env.BASE_URL + assetBaseUrl + src;
+}
